subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -45,6 +46,120 @@ static void oacp_l2cap_closed(struct bt_gatt_ots_l2cap *l2cap_ctx,
 }
 #endif
 
+#if defined(CONFIG_BT_OTS_OACP_CREATE_SUPPORT)
+static enum bt_gatt_ots_oacp_res_code oacp_create_proc_validate(
+	struct bt_conn *conn,
+	struct bt_ots *ots,
+	struct bt_gatt_ots_oacp_proc *proc)
+{
+	char str[BT_UUID_STR_LEN];
+	int err;
+	struct bt_gatt_ots_object *obj;
+	const struct bt_ots_obj_add_param param = {
+		.size = proc->create_params.size,
+		.type = proc->create_params.type,
+	};
+
+	bt_uuid_to_str(&param.type.uuid, str, BT_UUID_STR_LEN);
+	LOG_DBG("Validating Create procedure with size: 0x%08X and "
+		"type: %s", param.size, log_strdup(str));
+
+	if (!BT_OTS_OACP_GET_FEAT_CREATE(ots->features.oacp)) {
+		LOG_DBG("Create Procedure is not supported.");
+		return BT_GATT_OTS_OACP_RES_OPCODE_NOT_SUP;
+	}
+
+	err = bt_ots_obj_add_internal(ots, conn, &param, &obj);
+	if (err) {
+		goto exit;
+	}
+
+	/* Verify Initialization Metadata */
+	if (strlen(obj->metadata.name) > 0) {
+		LOG_ERR("Object name shall be a zero length string after object creation.");
+		(void)bt_ots_obj_delete(ots, obj->id);
+		err = -ECANCELED;
```

Comment:
Given the multiple additions of using `ECANCELED` it feels odd that the switch case does not have a `case` for `ECANCELED`.